examples/summarization/bart/evaluate_cnn.py

```diff
@@ -19,42 +20,83 @@ def chunks(lst, n):
 def generate_summaries(
     examples: list, out_file: str, model_name: str, batch_size: int = 8, device: str = DEFAULT_DEVICE
 ):
-    fout = Path(out_file).open("w")
-    model = BartForConditionalGeneration.from_pretrained(model_name).to(device)
-    tokenizer = BartTokenizer.from_pretrained("facebook/bart-large")
+    fout = Path(out_file).open("w", encoding="utf-8")
+    if "t5" in model_name:
+        model = T5ForConditionalGeneration.from_pretrained(model_name).to(device)
+        tokenizer = T5Tokenizer.from_pretrained(model_name)
+        # update config with summarization specific params
+        task_specific_params = model.config.task_specific_params
+        if task_specific_params is not None:
+            model.config.update(task_specific_params.get("summarization", {}))
+    else:
+        model = BartForConditionalGeneration.from_pretrained(model_name).to(device)
+        tokenizer = BartTokenizer.from_pretrained("facebook/bart-large")
 
     max_length = 140
     min_length = 55
 
     for batch in tqdm(list(chunks(examples, batch_size))):
+        if "t5" in model_name:
+            batch = [model.config.prefix + text for text in batch]
         dct = tokenizer.batch_encode_plus(batch, max_length=1024, return_tensors="pt", pad_to_max_length=True)
-        summaries = model.generate(
-            input_ids=dct["input_ids"].to(device),
-            attention_mask=dct["attention_mask"].to(device),
-            num_beams=4,
-            length_penalty=2.0,
-            max_length=max_length + 2,  # +2 from original because we start at step=1 and stop before max_length
-            min_length=min_length + 1,  # +1 from original because we start at step=1
-            no_repeat_ngram_size=3,
-            early_stopping=True,
-            decoder_start_token_id=model.config.eos_token_id,
-        )
+
+        input_ids = dct["input_ids"].to(device)
+        attention_mask = dct["attention_mask"].to(device)
+        if "t5" not in model_name:
+            summaries = model.generate(
```

Comment:
bart can use the config here just like `T5`

Reply:
Agreed!
However, the default config is different from what is supplied. Since people might want to compare Rouge results hence avoided overriding the defaults unless needed

Reply:
I added these supplied kwargs before the config was correct, so we can just use the config now.

Reply:
Done!